src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -61,6 +61,19 @@
  * <pre>
  * &lt;module name=&quot;NoWhitespaceBefore&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * int c = a ++; // violation: '++' is preceded with whitespace
+ * int c = a++; // OK
+ * int foo
+ * ; // violation: allowLineBreaks is set false by default
+ * int foo; // OK
+ * public int b ; // violation: ';' is preceded with whitespace
+ * public int b;  // OK
```

Comment:
similar as example with foo variable, can be third one there. No need for public modifier

Also, please consider using different variable names to make code look more valid